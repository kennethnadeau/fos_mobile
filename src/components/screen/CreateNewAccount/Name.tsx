import React, {FC} from 'react';
import RoundedButton from '@fos/components/RoundedButton';
import TextInput from '@fos/components/TextInput';
import {useTranslation} from 'react-i18next';
import {StyleSheet} from 'react-native';
import {s, vs} from 'react-native-size-matters';
import CarouselItemContainer from './CarouselItemContainer';
import FooterText from './FooterText';

export type NameProps = {
  firstName: string;
  lastName: string;
  onFirstNameChangeText: (name: string) => void;
  onLastNameChangeText: (name: string) => void;
  onFirstNameClear: () => void;
  onLastNameClear: () => void;
  onCreateUserPress: () => void;
  loading?: boolean;
};

const Name: FC<NameProps> = ({
  firstName,
  lastName,
  onFirstNameChangeText,
  onLastNameChangeText,
  onFirstNameClear,
  onLastNameClear,
  onCreateUserPress,
  loading = false,
}) => {
  const {t} = useTranslation();

  return (
    <CarouselItemContainer
      containerProps={{
        accessible: true,
        accessibilityLabel: t('screens.createNewAccount.enterNameHeader'),
      }}
      containerStyle={styles.carouselContainer}
      header={t('screens.createNewAccount.enterNameHeader')}>
      <TextInput
        accessibilityLabel={t('screens.createNewAccount.firstNameInput')}
        accessible
        autoCapitalize="words"
        containerStyle={styles.nameInput}
        onChangeText={onFirstNameChangeText}
        onClear={onFirstNameClear}
        placeholder={t('screens.createNewAccount.firstNameInput')}
        value={firstName}
      />

      <TextInput
        accessibilityLabel={t('screens.createNewAccount.lastNameInput')}
        accessible
        containerStyle={styles.nameInput}
        onChangeText={onLastNameChangeText}
        onClear={onLastNameClear}
        placeholder={t('screens.createNewAccount.lastNameInput')}
        value={lastName}
      />

      <RoundedButton
        accessibilityLabel={t(
          'screens.createNewAccount.enterNameButton.createUser',
        )}
        accessible
        disabled={!firstName || !lastName || loading}
        loading={loading}
        onPress={onCreateUserPress}
        size="large"
        title={t('screens.createNewAccount.enterNameButton.createUser')}
      />

      <FooterText style={styles.footer}>
        {t('screens.createNewAccount.enterNameFooter')}
      </FooterText>
    </CarouselItemContainer>
  );
};

const styles = StyleSheet.create({
  carouselContainer: {
    paddingHorizontal: s(28),
  },
  footer: {marginTop: vs(16), textAlign: 'center'},
  nameInput: {
    paddingHorizontal: 0,
  },
});

export default Name;
