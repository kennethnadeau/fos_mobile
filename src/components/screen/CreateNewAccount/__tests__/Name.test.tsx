import React from 'react';
import {render, RenderAPI} from '@fos/utils/testing';
import Name, {NameProps} from '../Name';
import enUs from '@fos/locale/languages/en-US';

const localization = enUs.translation.screens.createNewAccount;

describe('<Name />', () => {
  let name: RenderAPI;
  const props: NameProps = {
    onFirstNameClear: jest.fn(),
    onLastNameClear: jest.fn(),
    firstName: 'John',
    lastName: 'Doe',
    onCreateUserPress: jest.fn(),
    onFirstNameChangeText: jest.fn(),
    onLastNameChangeText: jest.fn(),
  };

  beforeEach(() => {
    name = render(<Name {...props} />);
  });

  it('should show header', () => {
    const actual = name.queryByText(localization.enterNameHeader);

    expect(actual).not.toBeNull();
  });

  it('should show first name input', () => {
    const actual = name.queryByA11yLabel(localization.firstNameInput);

    const placeholder = name.queryByPlaceholder(localization.firstNameInput);

    expect(actual).not.toBeNull();
    expect(placeholder).not.toBeNull();
  });

  it('should show last name input', () => {
    const actual = name.queryByA11yLabel(localization.lastNameInput);

    const placeholder = name.queryByPlaceholder(localization.lastNameInput);

    expect(actual).not.toBeNull();
    expect(placeholder).not.toBeNull();
  });

  it('should show create user button', () => {
    const actual = name.queryByA11yLabel(
      localization.enterNameButton.createUser,
    );

    expect(actual).not.toBeNull();
  });

  it('should show footer', () => {
    const actual = name.queryByText(localization.enterNameFooter);

    expect(actual).not.toBeNull();
  });

  it('should match snapshot', () => {
    expect(name.toJSON()).toMatchSnapshot();
  });
});
