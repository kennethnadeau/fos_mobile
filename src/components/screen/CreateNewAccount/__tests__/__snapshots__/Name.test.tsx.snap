// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Name /> should match snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    accessibilityLabel="Enter Name"
    accessible={true}
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "paddingHorizontal": 85.71428571428571,
        },
        Object {
          "paddingHorizontal": 60,
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 27.5,
          "marginVertical": 39.23529411764706,
        }
      }
      theme={
        Object {
          "Button": Object {
            "buttonStyle": Array [
              Object {
                "backgroundColor": "#4CA1D6",
              },
            ],
            "titleStyle": Array [
              Object {
                "color": "white",
                "fontSize": 34.285714285714285,
              },
            ],
          },
          "Icon": Object {
            "color": "#2089dc",
          },
          "Input": Object {
            "inputContainerStyle": Object {
              "alignItems": "center",
              "borderBottomWidth": 0.5,
              "borderColor": "#fff",
              "borderRadius": 64.28571428571428,
              "borderWidth": 0.5,
              "overflow": "hidden",
              "padding": 9.428571428571429,
            },
            "inputStyle": Object {
              "alignItems": "center",
              "color": "#fff",
              "padding": 9.428571428571429,
            },
            "placeholderTextColor": "#7E8395",
          },
          "Text": Object {
            "style": Object {
              "color": "white",
              "fontSize": 34.285714285714285,
            },
          },
          "colors": Object {
            "disabled": "hsl(208, 8%, 90%)",
            "divider": "#bcbbc1",
            "error": "#ff190c",
            "grey0": "#393e42",
            "grey1": "#43484d",
            "grey2": "#5e6977",
            "grey3": "#86939e",
            "grey4": "#bdc6cf",
            "grey5": "#e1e8ee",
            "greyOutline": "#bbb",
            "platform": Object {
              "android": Object {
                "error": "#f44336",
                "primary": "#2196f3",
                "secondary": "#9C27B0",
                "success": "#4caf50",
                "warning": "#ffeb3b",
              },
              "ios": Object {
                "error": "#ff3b30",
                "primary": "#007aff",
                "secondary": "#5856d6",
                "success": "#4cd964",
                "warning": "#ffcc00",
              },
            },
            "primary": "#2089dc",
            "searchBg": "#303337",
            "secondary": "#8F0CE8",
            "success": "#52c41a",
            "warning": "#faad14",
          },
        }
      }
    >
      Enter Name
    </Text>
    <View
      style={
        Object {
          "paddingHorizontal": 0,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "borderBottomWidth": 0.5,
            "borderColor": "#fff",
            "borderRadius": 64.28571428571428,
            "borderWidth": 0.5,
            "flexDirection": "row",
            "overflow": "hidden",
            "padding": 9.428571428571429,
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <TextInput
          accessibilityLabel="First Name"
          accessible={true}
          allowFontScaling={true}
          autoCapitalize="words"
          editable={true}
          onChangeText={[MockFunction]}
          placeholder="First Name"
          placeholderTextColor="#7E8395"
          rejectResponderTermination={true}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "color": "#fff",
              "flex": 1,
              "fontSize": 18,
              "minHeight": 40,
              "padding": 9.428571428571429,
            }
          }
          testID="RNE__Input__text-input"
          underlineColorAndroid="transparent"
          value="John"
        />
        <View
          style={
            Object {
              "alignItems": "center",
              "height": 40,
              "justifyContent": "center",
              "marginVertical": 4,
              "paddingRight": 4,
            }
          }
        >
          <View
            style={
              Object {
                "overflow": "hidden",
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                  }
                }
              >
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": "#7E8395",
                        "fontSize": 24,
                      },
                      Object {
                        "backgroundColor": "transparent",
                      },
                      Object {
                        "fontFamily": "FontAwesome",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                  testID="iconIcon"
                >
                  
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <Text
        style={
          Object {
            "color": "#ff190c",
            "fontSize": 12,
            "margin": 5,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "paddingHorizontal": 0,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "borderBottomWidth": 0.5,
            "borderColor": "#fff",
            "borderRadius": 64.28571428571428,
            "borderWidth": 0.5,
            "flexDirection": "row",
            "overflow": "hidden",
            "padding": 9.428571428571429,
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <TextInput
          accessibilityLabel="Last Name"
          accessible={true}
          allowFontScaling={true}
          editable={true}
          onChangeText={[MockFunction]}
          placeholder="Last Name"
          placeholderTextColor="#7E8395"
          rejectResponderTermination={true}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "color": "#fff",
              "flex": 1,
              "fontSize": 18,
              "minHeight": 40,
              "padding": 9.428571428571429,
            }
          }
          testID="RNE__Input__text-input"
          underlineColorAndroid="transparent"
          value="Doe"
        />
        <View
          style={
            Object {
              "alignItems": "center",
              "height": 40,
              "justifyContent": "center",
              "marginVertical": 4,
              "paddingRight": 4,
            }
          }
        >
          <View
            style={
              Object {
                "overflow": "hidden",
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                  }
                }
              >
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": "#7E8395",
                        "fontSize": 24,
                      },
                      Object {
                        "backgroundColor": "transparent",
                      },
                      Object {
                        "fontFamily": "FontAwesome",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                  testID="iconIcon"
                >
                  
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <Text
        style={
          Object {
            "color": "#ff190c",
            "fontSize": 12,
            "margin": 5,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "borderRadius": 3,
          "overflow": "hidden",
          "width": "100%",
        }
      }
    >
      <View
        accessibilityLabel="Create User"
        accessibilityRole="button"
        accessibilityState={
          Object {
            "busy": false,
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#4CA1D6",
              "borderColor": "#2089dc",
              "borderRadius": 85.71428571428571,
              "borderWidth": 0,
              "flexDirection": "row",
              "height": 107.14285714285714,
              "justifyContent": "center",
              "padding": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 34.285714285714285,
                "paddingBottom": 1,
                "paddingTop": 2,
                "textAlign": "center",
              }
            }
          >
            Create User
          </Text>
        </View>
      </View>
    </View>
    <Text
      style={
        Object {
          "color": "#7E8395",
          "fontSize": 34.285714285714285,
          "marginTop": 31.388235294117646,
          "textAlign": "center",
        }
      }
      theme={
        Object {
          "Button": Object {
            "buttonStyle": Array [
              Object {
                "backgroundColor": "#4CA1D6",
              },
            ],
            "titleStyle": Array [
              Object {
                "color": "white",
                "fontSize": 34.285714285714285,
              },
            ],
          },
          "Icon": Object {
            "color": "#2089dc",
          },
          "Input": Object {
            "inputContainerStyle": Object {
              "alignItems": "center",
              "borderBottomWidth": 0.5,
              "borderColor": "#fff",
              "borderRadius": 64.28571428571428,
              "borderWidth": 0.5,
              "overflow": "hidden",
              "padding": 9.428571428571429,
            },
            "inputStyle": Object {
              "alignItems": "center",
              "color": "#fff",
              "padding": 9.428571428571429,
            },
            "placeholderTextColor": "#7E8395",
          },
          "Text": Object {
            "style": Object {
              "color": "white",
              "fontSize": 34.285714285714285,
            },
          },
          "colors": Object {
            "disabled": "hsl(208, 8%, 90%)",
            "divider": "#bcbbc1",
            "error": "#ff190c",
            "grey0": "#393e42",
            "grey1": "#43484d",
            "grey2": "#5e6977",
            "grey3": "#86939e",
            "grey4": "#bdc6cf",
            "grey5": "#e1e8ee",
            "greyOutline": "#bbb",
            "platform": Object {
              "android": Object {
                "error": "#f44336",
                "primary": "#2196f3",
                "secondary": "#9C27B0",
                "success": "#4caf50",
                "warning": "#ffeb3b",
              },
              "ios": Object {
                "error": "#ff3b30",
                "primary": "#007aff",
                "secondary": "#5856d6",
                "success": "#4cd964",
                "warning": "#ffcc00",
              },
            },
            "primary": "#2089dc",
            "searchBg": "#303337",
            "secondary": "#8F0CE8",
            "success": "#52c41a",
            "warning": "#faad14",
          },
        }
      }
    >
      Enter name footer text
    </Text>
  </View>
</RNCSafeAreaProvider>
`;
